zipline/assets/asset_writer.py

```diff
@@ -0,0 +1,744 @@
+import pandas as pd
+import numpy as np
+from pandas.tseries.tools import normalize_date
+from six import with_metaclass, string_types
+from abc import (
+    ABCMeta,
+    abstractmethod,
+)
+from zipline.errors import (
+    ConsumeAssetMetaDataError,
+    InvalidAssetType,
+    SidAssignmentError,
+)
+from zipline.assets import (
+    Asset, Equity, Future
+)
+
+ASSET_FIELDS = [
```

Comment:
Order shouldn't matter, but can frozensets be combined in the same manner as I do here?